cmd/beemo/notify_mentions.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	appbsky "github.com/bluesky-social/indigo/api/bsky"
+	"github.com/bluesky-social/indigo/atproto/identity"
+	"github.com/bluesky-social/indigo/atproto/syntax"
+
+	"github.com/urfave/cli/v2"
+)
+
+type MentionChecker struct {
+	slackWebhookURL string
+	mentionDIDs     []syntax.DID
+	logger          *slog.Logger
+	directory       identity.Directory
+}
+
+func (mc *MentionChecker) ProcessPost(ctx context.Context, did syntax.DID, rkey syntax.RecordKey, post appbsky.FeedPost) error {
+	mc.logger.Debug("processing post record", "did", did, "rkey", rkey)
+
+	for _, facet := range post.Facets {
+		for _, feature := range facet.Features {
+			mention := feature.RichtextFacet_Mention
+			if mention == nil {
+				continue
+			}
+			for _, d := range mc.mentionDIDs {
```

Comment:
Instead of a for loop over a list here, you could make `mentionDIDs` a map and key into it, would be a bit more idiomatic Go if that's the only way we're using that struct (to check membership), turn `mentionDIDs` into a `map[string]struct{}` which is Go's version of a "set"

Reply:
you are correct, but i'm going to be lazy and skip on this refactor in this case. i'll try to use that pattern more in the future